axum/src/routing/method_routing.rs

```diff
@@ -1034,13 +1034,11 @@ where
                     match $svc {
                         MethodEndpoint::None => {}
                         MethodEndpoint::Route(route) => {
-                            return RouteFuture::from_future(route.clone().oneshot_inner($req))
-                                .strip_body($method == Method::HEAD);
+                            return route.clone().oneshot_inner($req);
```

Comment:
I see that this is the same as https://github.com/tokio-rs/axum/pull/2897.
Do you think we should first merge https://github.com/tokio-rs/axum/pull/2897 to reduce the scope of this PR?

Reply:
Both PRs benefit from this change, so it can be merged in either order!